std/range/package.d

```diff
@@ -4435,6 +4435,37 @@ enum StoppingPolicy
     requireSameLength,
 }
 
+///
+pure @safe unittest
+{
+    import std.algorithm.comparison : equal;
+    import std.exception : assertThrown;
+    import std.range.primitives;
+    import std.typecons : tuple;
+
+    auto a = [1, 2, 3];
+    auto b = [4, 5, 6, 7];
+
+    auto shortest = zip(StoppingPolicy.shortest, a.save, b.save);
```

Comment:
No need for `.save` here. Arrays are views only and the ptr + length is copied automatically. Also `zip` doesn't use `ref`